tests/testthat/test-conversion.R

```diff
@@ -255,3 +255,151 @@ test_that("as_adjacency_matrix() works -- dense + not both", {
     )
   )
 })
+
+test_that("as_edgelist() works", {
+  g <- sample_gnp(100, 3 / 100)
+  e <- as_edgelist(g)
+  g2 <- make_graph(t(e), n = vcount(g), dir = FALSE)
+  expect_isomorphic(g, g2)
+})
+
+test_that("as_adj_list() works", {
+  g <- sample_gnp(50, 2 / 50)
+  al <- as_adj_list(g)
+  g2 <- graph_from_adj_list(al, mode = "all")
+  expect_isomorphic(g, g2)
+  expect_true(isomorphic(g, g2,
+    vertex.color1 = 1:vcount(g),
+    vertex.color2 = 1:vcount(g2),
+    method = "vf2"
+  ))
+
+  ####
+
+  el <- as_adj_edge_list(g)
+  for (i in 1:vcount(g)) {
+    a <- E(g)[.inc(i)]
+    expect_equal(length(a), length(el[[i]]), ignore_attr = TRUE)
+    expect_equal(sort(el[[i]]), sort(a), ignore_attr = TRUE)
+  }
+
+  g <- sample_gnp(50, 4 / 50, directed = TRUE)
+  el1 <- as_adj_edge_list(g, mode = "out")
+  el2 <- as_adj_edge_list(g, mode = "in")
+  for (i in 1:vcount(g)) {
+    a <- E(g)[.from(i)]
+    expect_equal(length(a), length(el1[[i]]), ignore_attr = TRUE)
+    expect_equal(sort(el1[[i]]), sort(a), ignore_attr = TRUE)
+  }
+  for (i in 1:vcount(g)) {
+    a <- E(g)[.to(i)]
+    expect_equal(length(a), length(el2[[i]]), ignore_attr = TRUE)
+    expect_equal(sort(el2[[i]]), sort(a), ignore_attr = TRUE)
+  }
+})
+
+test_that("graph_from_graphnel() works", {
+  skip_if_not_installed("graph")
+  suppressPackageStartupMessages(library(graph, warn.conflicts = FALSE))
+
+  g <- sample_gnp(100, 5 / 100)
+  N <- as_graphnel(g)
+  g2 <- graph_from_graphnel(N)
+  gi <- graph.isomorphic.vf2(g, g2)
+  expect_true(gi$iso)
+  expect_equal(gi$map12, 1:vcount(g))
+  expect_equal(gi$map21, 1:vcount(g))
+
+  ## Attributes
+
+  V(g)$name <- as.character(vcount(g):1)
+  E(g)$weight <- sample(1:10, ecount(g), replace = TRUE)
+  g$name <- "Foobar"
+
+  N <- as_graphnel(g)
+  g2 <- graph_from_graphnel(N)
+  expect_isomorphic(g, g2)
+  expect_equal(V(g)$name, V(g2)$name)
+
+  A <- as_adjacency_matrix(g, attr = "weight", sparse = FALSE)
+  A2 <- as_adjacency_matrix(g2, attr = "weight", sparse = FALSE)
+  expect_equal(A, A)
+  expect_equal(g$name, g2$name)
+})
+
+test_that("as_graphnel() does not duplicate loop edges", {
+  skip_if_not_installed("graph")
+
+  mat <- matrix(c(1, 0.5, 0.5, 0), nrow = 2)
+  dimnames(mat) <- list(c("A", "B"), c("A", "B"))
+
+  igr <- graph_from_adjacency_matrix(mat, mode = "undirected", weighted = TRUE)
+
+  grNEL <- as_graphnel(igr)
+  expect_equal(graph::edgeL(grNEL)$A$edges, c(1, 2))
+})
+
+
+test_that("as_long_data_frame() works correctly with and without names", {
+  expect_snapshot({
+    ring <- make_ring(3)
+    as_long_data_frame(ring)
+
+    V(ring)$name <- letters[1:3]
+    as_long_data_frame(ring)
+
+    V(ring)$score <- LETTERS[1:3]
+    as_long_data_frame(ring)
+
+    E(ring)$info <- 3:1
+    as_long_data_frame(ring)
+  })
+})
+
+test_that("as_biadjacency_matrix() works -- dense", {
+  I <- matrix(sample(0:1, 35, replace = TRUE, prob = c(3, 1)), ncol = 5)
```

Comment:
Minor nitpick: Let's not call this `I`, as that's the standard name for the identity matrix. (It was probably called `I` because of "incidence" matrix.)

Reply:
and good to merge once I fix that?

Reply:
This part is good, and this is just an aesthetic nitpick that doesn't necessarily need to be fixed.  My main comments are in the other thread in this PR, addressing string matrices. Will open an issue for that soon.